compiler/src/dotty/tools/dotc/core/classfile/ClassfileParser.scala

```diff
@@ -917,8 +917,13 @@ class ClassfileParser(
               }
             case _ =>
               if (classfile.jpath == null) {
-                report.error("Could not load TASTY from .tasty for virtual file " + classfile)
-                Array.empty
+                val tastyFileOrNull = classfile.container
+                  .lookupName(classfile.name.stripSuffix(".class") + ".tasty", false)
```

Comment:
Does this need to be a special-case just for virtual files or could this codepath also work for plain files?

Reply:
This should work for plain files too, I think. Using it there could be a way to test that, actually…